### DISCUSSION TITLE
Interval equals in case of timezones

### DISCUSSION THREAD

**Discussion by dmitrijs-balcers** [moment/luxon#1349] [Q&A] [answered]
```
  const s = Interval.after(DateTime.fromISO("2022-09-19T00:00:00.000+03:00", { setZone: true }), { days: 1 });
  const s2 = Interval.after(DateTime.fromISO("2022-09-19T01:00:00.000+04:00", { setZone: true }), { days: 1 });

  expect(s.equals(s2)).toBeTruthy();
 ```

This should be a passing test, correct? (it doesn't at the moment)

I expected that equals compares two intervals regardless of timezones, so that it would tell me if two intervals are the same, regardless of timezones.

So I ended up using something like this instead in my project, to compare if two intervals are the same regardless of TZ:
```
const equals = (a: Interval) => (b: Interval) => a.engulfs(b) && b.engulfs(a);
```

**Comment by icambron** [comment, marked as answer]
An interval is equal to another interval if both their starts and ends are equal, and two datetimes are equal if they represent the same time in the same zone. So that nets out to your intervals not being equal.

I think this is the right behavior for intervals to have, because the two intervals will have different results for the same methods As a simple example, consider the methods that format intervals to strings. It would be weird if two objects that considered themselves equal had divergent behavior.

I would support adding a separate equality operator that only compared the epoch millis of the start and end. We don't have one for datetimes because it's easy enough to do `+d1 == +d2` but that's a pain for intervals.

You probably didn't want to use `setZone` here. Without it, Luxon will convert the zones to the system zone, taking the +3 and +4 into account while doing so. You could also specify `{zone: "utc" }`
